samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -0,0 +1,254 @@
+		ret = -ENOMEM;
+		LOG_ERR("Failed to append NMEA data to GNSS message object");
+		goto cleanup;
+	}
+
+	/* Send the sensor sample container object as a device message */
+	ret = send_device_message_cJSON(msg_obj);
+
+cleanup:
+	if (msg_obj) {
+		cJSON_Delete(msg_obj);
+	}
+	return ret;
+}
+
+
+/**
+ * @brief Compute checksum for NMEA string
+ *
+ * @param datastring - The NMEA string to compute the checksum for
+ * @return uint8_t - The resultant checksum
+ */
+static uint8_t nmea_checksum(const char *const datastring)
+{
+	uint8_t checksum = 0;
+	int len = strlen(datastring);
+
+	for (int i = 0; i < len; i++) {
+		checksum ^= datastring[i];
+	}
+	return checksum;
+}
+
+/**
+ * @brief Callback to receive periodic location updates from location_tracking.c and forward them
+ * to nRF Cloud.
+ *
+ * Note that cellular positioning (MCELL/Multi-Cell and SCELL/Single-Cell) is sent to nRF
+ * Cloud automatically (since the Location library and nRF Cloud must work together to
+ * determine them in the first place). GNSS positions, on the other hand, must be
+ * sent manually, since they are determined entirely on-device.
+ *
+ * @param location_data - The received location update.
+ *
+ */
+static void on_location_update(const struct location_data location_data)
+{
+	LOG_INF("Location Updated: %.06f N %.06f W, accuracy: %.01f m, Method: %s",
+		location_data.latitude, location_data.longitude, location_data.accuracy,
+		location_data.method == LOCATION_METHOD_CELLULAR	? "Cellular" :
+		location_data.method == LOCATION_METHOD_GNSS		? "GNSS"     :
+		location_data.method == LOCATION_METHOD_WIFI		? "WIFI"     :
+									  "Invalid");
+
+	/* If the position update was derived using GNSS, send it onward to nRF Cloud */
+	if (location_data.method == LOCATION_METHOD_GNSS) {
+		LOG_INF("GNSS Position Update! Sending to nRF Cloud...");
+
+		/* Synthesize an NMEA message from the provided lat/long/timestamp
+		 * This is necessary, because the nRF Cloud MQTT API only supports NMEA messages
+		 */
+
+		char nmea_buf[45];
+		int latdeg = floor(fabs(location_data.latitude));
+		int londeg = floor(fabs(location_data.longitude));
+		double latmin = (fabs(location_data.latitude) - latdeg) * 60;
+		double lonmin = (fabs(location_data.longitude) - londeg) * 60;
+
+		/* Not actually a valid NMEA,
+		 * but close enough that nRF Cloud can't tell the difference
+		 */
+		snprintf(nmea_buf, sizeof(nmea_buf), "$GPGGA,,%02d%05.2f,%c,%02d%05.2f,%c,,,,,,,,,",
+			latdeg, latmin, location_data.latitude > 0 ? 'N':'S',
+			londeg, lonmin, location_data.longitude > 0 ? 'E':'W');
+
+		int payloadlength = strlen(nmea_buf);
+
+		snprintf(nmea_buf+payloadlength, sizeof(nmea_buf)-payloadlength, "*%02X\n",
+			 nmea_checksum(nmea_buf + 1));
+		LOG_DBG("NMEA: %s", log_strdup(nmea_buf));
+
+		/* Send the NMEA string */
+		send_gnss_nmea(nmea_buf);
+	}
+}
+
+void main_application(void)
+{
+	/* Wait for the date and time to become known
```

Comment:
Add periods?

Reply:
Seems I missed a bunch of formatting corrections I thought I had applied. I'll do another once-over of all of the comments.

Reply:
Okay I think I've gotten all of the comments I missed. Thanks for catching this.